src/App.jsx

```diff
@@ -1,35 +1,29 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import React, { useState, useEffect } from 'react';
+import { BrowserRouter, Routes, Route} from 'react-router-dom';
+import { getArticles } from './api';
+import { Nav } from './components/Nav';
+import { Header } from './components/Header';
+import { Homepage } from './components/Homepage';
 
-function App() {
-  const [count, setCount] = useState(0)
+const App = () => {
+  const [filterTopic, setFilterTopic] = useState("");
 
   return (
-    <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo" />
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo" />
-        </a>
-      </div>
-      <h1>Vite + React</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.jsx</code> and save to test HMR
-        </p>
-      </div>
-      <p className="read-the-docs">
-        Click on the Vite and React logos to learn more
-      </p>
-    </>
-  )
-}
+    <BrowserRouter>
+      <div className="app">
+        <Header />
+        <Nav setFilterTopic={setFilterTopic} />
+        <Routes>
+          <Route path="/" element={<Homepage filterTopic={filterTopic} />}/>
+          
 
-export default App
+          {/* <Route path="/single-article/:id">
+            <SingleArticlePage articles={articles} />
+          </Route> */}
```

Comment:
Your PRs should be a fairly clean & polished version of the code, so when it comes to commented out sections I'd delete them - this ensures that your main branch is clean, since that's where merged code goes